src/Microsoft.ComponentDetection.Detectors/rust/Contracts/CargoMetadata.cs

```diff
@@ -453,8 +453,16 @@ public override CrateType Read(ref Utf8JsonReader reader, Type typeToConvert, Js
                 return CrateType.ProcMacro;
             case "test":
                 return CrateType.Test;
+            case "rlib":
+                return CrateType.RLib;
+            case "dylib":
+                return CrateType.DyLib;
+            case "cdylib":
+                return CrateType.CdyLib;
+            case "staticlib": 
+                return CrateType.StaticLib;
         }
-        throw new Exception("Cannot unmarshal type CrateType");
+        throw new Exception($"Cannot unmarshal type CrateType - {value}");
```

Comment:
Should we return type Other and log this as an error? Guess it depends if it would break things / how often new CrateTypes are added / how many we might not know about.

Reply:
I used the docs to determine the full set of allowable values, and it doesn't look like this is something that's updated often, My only worry with just logging it as an error is that if something new does get introduced in the future we would just silently be marking those as unknown when there could be some additional processing that needs to happen for a potentially new crate-type